Description: Do not look for optional packages when running the testsuite
Origin: https://git.archlinux.org/svntogit/community.git/tree/trunk?h=packages/sagemath

--- a/sage/src/sage/doctest/control.py
+++ b/sage/src/sage/doctest/control.py
@@ -325,22 +325,6 @@
             if "all" in options.optional:
                 # Special case to run all optional tests
                 options.optional = True
-            else:
-                # We replace the 'optional' tag by all optional
-                # packages for which the installed version matches the
-                # latest available version (this implies in particular
-                # that the package is actually installed).
-                if 'optional' in options.optional:
-                    options.optional.discard('optional')
-                    from sage.misc.package import list_packages
-                    for pkg in list_packages('optional', local=True).values():
-                        if pkg['installed_version'] == pkg['remote_version']:
-                            options.optional.add(pkg['name'])
-
-                # Check that all tags are valid
-                for o in options.optional:
-                    if not optionaltag_regex.search(o):
-                        raise ValueError('invalid optional tag {!r}'.format(o))
 
         self.options = options
         self.files = args
--- a/sage/src/sage/doctest/test.py
+++ b/sage/src/sage/doctest/test.py
@@ -463,14 +463,6 @@
     ...
     0
 
-Test an invalid value for ``--optional``::
-
-    sage: subprocess.call(["sage", "-t", "--warn-long", "0", "--optional=bad-option", "show_skipped.rst"], **kwds)
-    Traceback (most recent call last):
-    ...
-    ValueError: invalid optional tag 'bad-option'
-    1
-
 Test ``atexit`` support in the doctesting framework::
 
     sage: F = tmp_filename()
